desktop-src/direct3dhlsl_current/hlsl-method-gatherCmp.md

```diff
@@ -0,0 +1,283 @@
+      in  SamplerState  samp,
+      in  float3        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+```
+
+#### TextureCubeArray
+```syntax
+// same as GatherRed
+ElementType TextureCubeArray<ElementType>::GatherCmp(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCubeArray<ElementType>::GatherCmpRed(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCubeArray<ElementType>::GatherCmpGreen(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCubeArray<ElementType>::GatherCmpBlue(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+
+ElementType TextureCubeArray<ElementType>::GatherCmpAlpha(
+      in  SamplerState  samp,
+      in  float4        coord,
+      in  float         compareValue,
+   [, out uint          status ] );
+```
+
+## Return Value
+
+The texture's template type for the element, which may be a scalar or vector.
+This provides the HLSL type for an element loaded from a texture.
+The format stored in the resource is based on the texture resource view's DXGI_FORMAT,
+which will determine how data is translated into the type specified in HLSL.
+
+## Parameters
+
+
+| Parameter | Description |
+| - | - |
+| in [`SamplerState samp`](#samplerstate-samp) | SamplerState object that determines how sampling will take place. |
+| in [`<CoordType> coord`](#coordtype-coord) | The texture coordinates. |
+| in [`float compareValue`](#float-comparevalue) | A value to compare each against each sampled value. |
+| in [`<OffsetType> offset`](#offsettype-offset) | The offset applied to the texture coordinates before sampling. |
+| \[ out [`uint status`](#uint-status) \] | The status of the operation. You can't access the status directly; instead, pass the status to the CheckAccessFullyMapped intrinsic function. CheckAccessFullyMapped returns TRUE if all values from the corresponding Sample, Gather, or Load operation accessed mapped tiles in a tiled resource. If any values were taken from an unmapped tile, CheckAccessFullyMapped returns FALSE. |
+
+Types that depend on texture object:
+
+| TextureObject | [`<CoordType>`](#coordtype-coord) | [`<OffsetType>`](#offsettype-offset) |
+| --- | --- | --- |
+| `Texture2D` | `float2` | `int2` |
+| `Texture2DArray` | `float3` | `int2` |
+| `TextureCube` | `float3` | N/A |
+| `TextureCubeArray` | `float4` | N/A |
+
+### `SamplerState samp`
+
+A Sampler state. This is an object declared in an effect file that contains state assignments.
+
+> My alternate text:
+> SamplerState object that determines how sampling will take place.
```

Comment:
Should have resolved these and "TBD" items instead of replicating them elsewhere.

Reply:
Great alternate text here.

Reply:
Replaced all instances with links added.